src/build.d

```diff
@@ -442,6 +449,55 @@ auto dmdUnittestExe = dmdExe("-unittest", ["-version=NoMain", "-unittest", "-mai
         .command(dmdUnittestExe.targets);
 });
 
+/**
+BuildRule to run the DMD frontend header generation
+For debugging, use `./build.d cxx-headers DFLAGS="-debug=Debug_DtoH"` (clean before)
+*/
+alias buildFrontendHeaders = makeRule!((builder, rule) {
+    const dmdSources = sources.dmd.all ~ sources.root ~ sources.lexer;
+    const dmdExeFile = dmdDefault.deps[0].target;
+    builder
+        .name("cxx-headers")
+        .description("Build the C++ frontend headers ")
+        .msg("(RUN) CXX-HEADERS")
+        .deps([dmdDefault])
+        .target(env["G"].buildPath("frontend.h"))
+        .command([dmdExeFile] ~ flags["DFLAGS"] ~
+            ["-J" ~ env["RES"], "-c", "-o-", "-HCf="~rule.target] ~ dmdSources);
+});
+
+alias runCxxHeadersTest = makeRule!((builder, rule) {
+    builder
+        .name("cxx-headers-test")
+        .description("Test the C++ frontend headers ")
+        .msg("(TEST) CXX-HEADERS")
+        .deps([buildFrontendHeaders])
+        .commandFunction(() {
+            const cxxHeaderGeneratedPath = buildFrontendHeaders.target;
+            const cxxHeaderReferencePath = env["D"].buildPath("frontend.h");
+            log("Comparing referenceHeader(%s) <-> generatedHeader(%s)",
+                cxxHeaderReferencePath, cxxHeaderGeneratedPath);
+            const generatedHeader = cxxHeaderGeneratedPath.readText;
+            const referenceHeader = cxxHeaderReferencePath.readText;
+            if (generatedHeader != referenceHeader) {
```

Comment:
Maybe this should use `git diff` by default given the size of `frontend.h` and that it allows to ignore small whitespace changes?

Reply:
Hmm, this would make the test depended on `git` which might not be available when the archives are downloaded as a zip.

Reply:
Fair enough, allthough we already have certain targets that depend on `git`, (e.g. style checks)

Reply:
Yes, but these targets are optional and a contributor doesn't need to run them locally, whereas there's a good chance that he might have to regenerate the header file.

Reply:
True. Just having a hard time imaging someone trying to contribute without having a local `git` installation.

Anyway, that was just an idea because reading both files might be quite memory expensive.

Reply:
The file is currently 530K - I doubt that people will complain about one 1MB memory usage. We can always switch to `git diff` fully if this is really an issue?

(though I don't have strong feelings either way, so I would be happy to change)

Reply:
Don't think so either, was more thinking about the possibility to ignore small whitespace changes. But let's give this a shot and change it if it becomes an actual problem/annoyance